File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        
        int n = queue1.length;

        long current = 0;
        long total = 0;

        Queue<Integer> qu1 = new ArrayDeque<>();
        Queue<Integer> qu2 = new ArrayDeque<>();

        for (int i = 0; i < n; i++) {
            current += queue1[i];
            total += queue1[i] + queue2[i];

            qu1.offer(queue1[i]);
            qu2.offer(queue2[i]);
        }

        if (total % 2 != 0) return -1;

        long target = total / 2;

        int ans = 0;
        while (ans <= n * 4) {

            if (current == target) return ans;

            if (current < target) {
                current += qu2.peek();
                qu1.offer(qu2.poll());
            } else {
                current -= qu1.peek();
                qu2.offer(qu1.poll());
            }

            ans++;
        }

        return -1;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque


def solution(queue1, queue2):

    n = len(queue1)

    current = 0
    total = 0

    qu1 = deque(queue1)
    qu2 = deque(queue2)

    for i in range(n):
        current += queue1[i]
        total += queue1[i] + queue2[i]

    if total % 2 != 0:
        return -1

    target = total // 2
    ans = 0

    while ans <= n * 4:

        if current == target:
            return ans

        if current < target:
            current += qu2[0]
            qu1.append(qu2.popleft())
        else:
            current -= qu1[0]
            qu2.append(qu1.popleft())

        ans += 1

    return -1
